### PR TITLE
Feature/fcla 92 split project into subprojects

### PR DESCRIPTION
**Pull request by sbraggion**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -13,83 +13,98 @@ plugins {
...
+    apply(plugin = "org.danilopianini.git-sensitive-semantic-versioning")
+    apply(plugin = "org.gradle.jacoco")
+    apply(plugin = "pl.droidsonroids.jacoco.testkit")
+    apply(plugin = "org.jlleitschuh.gradle.ktlint")
+    apply(plugin = "io.gitlab.arturbosch.detekt")
+    apply(plugin = "org.jetbrains.dokka")
+
+    afterEvaluate {
+        tasks.jar {
+            // Otherwise it throws a "No main manifest attribute" error
```
Comment (nlasagni) [inline build.gradle.kts]: This maybe should be placed above line 38?

File: build.gradle.kts [hunk 2]
```diff
@@ -13,83 +13,98 @@ plugins {
...
+                attributes(
+                    mapOf(
+                        "Main-Class" to ext.get(mainClassVarName),
+                        "Implementation-Version" to archiveVersion
+                    )
+                )
+            }
+            // All the dependencies needed by our application doesn't exists in the same classpath where our .jar resides.
+            // In order to make our application standalone, we must include all of its dependencies inside our application.
+            // Include all of its dependencies inside our application
```
Comment (nlasagni) [inline build.gradle.kts]: This is already included in the comment at line 47?

File: build.gradle.kts [hunk 3]
```diff
@@ -13,83 +13,98 @@ plugins {
...
+
+            // To add all of the dependencies otherwise a "NoClassDefFoundError" error
+            from(sourceSets.main.get().output)
+            dependsOn(configurations.runtimeClasspath)
+
+            // Configurations.compile is a reference to the all of the artifacts defined in the compile configuration.
+            from({
+                // Wrap each of the JAR files with the zipTree() method. The result is a collection of ZIP file trees
+
+                // configurations.runtimeClasspath.get().filter { it.name.endsWith("jar") }.map { zipTree(it) }
```
Comment (nlasagni) [inline build.gradle.kts]: Is this commented line still useful?

File: build.gradle.kts [hunk 4]
```diff
@@ -13,83 +13,98 @@ plugins {
...
+                        "Main-Class" to ext.get(mainClassVarName),
+                        "Implementation-Version" to archiveVersion
+                    )
+                )
+            }
+            // All the dependencies needed by our application doesn't exists in the same classpath where our .jar resides.
+            // In order to make our application standalone, we must include all of its dependencies inside our application.
+            // Include all of its dependencies inside our application
+
+            // To add all of the dependencies otherwise a "NoClassDefFoundError" error
```
Comment (nlasagni) [inline build.gradle.kts]: This comment has no useful meaning, maybe should be changed properly.

**Review by nlasagni — CHANGES_REQUESTED** [review]

**Review by nlasagni — DISMISSED** [review]

File: reservation/src/main/kotlin/it/unibo/lss/fcla/reservation/domain/entities/exceptions/ConsultingReservationFreelancerCannotBeEmpty.kt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package it.unibo.lss.fcla.reservation.domain.entities.exceptions
```
Comment (Giordo94) [inline reservation/src/main/kotlin/it/unibo/lss/fcla/reservation/domain/entities/exceptions/ConsultingReservationFreelancerCannotBeEmpty.kt]: This is the old version. Delete this file

File: reservation/src/main/kotlin/it/unibo/lss/fcla/reservation/domain/entities/exceptions/ClosedReservationCannotBeUpdated.kt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package it.unibo.lss.fcla.reservation.domain.entities.exceptions
```
Comment (Giordo94) [inline reservation/src/main/kotlin/it/unibo/lss/fcla/reservation/domain/entities/exceptions/ClosedReservationCannotBeUpdated.kt]: This is the old version. Delete this file

File: reservation/src/main/kotlin/it/unibo/lss/fcla/reservation/domain/entities/exceptions/WorkoutReservationAimCannotBeEmpty.kt [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package it.unibo.lss.fcla.reservation.domain.entities.exceptions
```
Comment (Giordo94) [inline reservation/src/main/kotlin/it/unibo/lss/fcla/reservation/domain/entities/exceptions/WorkoutReservationAimCannotBeEmpty.kt]: This is the old version. Delete this file

File: reservation/build.gradle.kts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+val myMainClass = "it.unibo.lss.fcla.MainClassKt"
```
Comment (Giordo94) [inline reservation/build.gradle.kts]: Main class was renamed. Please update with "it.unibo.lss.fcla.reservation.InteractiveReservationMicroserviceStarterClassKt"

**Review by Giordo94 — CHANGES_REQUESTED** [review]
Some small fix needed

**Review by Giordo94 — APPROVED** [review]
Approved

**Pull request merged**